repose-aggregator/services/datastore/impl/src/main/java/org/openrepose/core/services/datastore/impl/distributed/remote/command/AbstractRemoteCommand.java

```diff
@@ -57,8 +58,9 @@ public String getBaseUrl() {
     }
 
     protected Map<String, String> getHeaders(RemoteBehavior remoteBehavior) {
-        Map<String, String> headers = new HashMap<String, String>();
+        Map<String, String> headers = new HashMap<>();
         headers.put(DatastoreHeader.HOST_KEY.toString(), hostKey);
+        headers.put(CommonHttpHeader.TRACE_GUID.toString(), remoteEndpoint.getHostName());
```

Comment:
you should be using the same guid that is currently being run under on this node